packages/slate-react/src/components/editor.tsx

```diff
@@ -0,0 +1,510 @@
+      <ReadOnlyContext.Provider value={readOnly}>
+        <FocusedContext.Provider value={editor.isFocused()}>
+          <div
+            // COMPAT: The Grammarly Chrome extension works by changing the DOM
+            // out from under `contenteditable` elements, which leads to weird
+            // behaviors so we have to disable it like editor. (2017/04/24)
+            data-gramm={false}
+            role={readOnly ? undefined : props.role || 'textbox'}
+            {...attributes}
+            data-slate-editor
+            data-slate-node="value"
+            contentEditable={readOnly ? undefined : true}
+            suppressContentEditableWarning
+            ref={ref}
+            style={{
+              // Prevent the default outline styles.
+              outline: 'none',
+              // Preserve adjacent whitespace and new lines.
+              whiteSpace: 'pre-wrap',
+              // Allow words to break if they are too long.
+              wordWrap: 'break-word',
+              // Allow for passed-in styles to override anything.
+              ...style,
+            }}
+            onBlur={useCallback((event: React.FocusEvent) => {
+              if (
+                !readOnly &&
+                !state.isUpdatingSelection &&
+                hasEditableTarget(editor, event.target) &&
+                // COMPAT: If the current `activeElement` is still the previous
+                // one, this is due to the window being blurred when the tab
+                // itself becomes unfocused, so we want to abort early to allow to
+                // editor to stay focused when the tab becomes focused again.
+                state.latestElement !== window.document.activeElement
+              ) {
+                const { relatedTarget } = event
+
+                // COMPAT: The `relatedTarget` can be null when the new focus target
+                // is not a "focusable" element (eg. a `<div>` without `tabindex`
+                // set).
+                if (relatedTarget) {
+                  const el = editor.toDomNode(value)
+
+                  // COMPAT: The event should be ignored if the focus is returning
+                  // to the editor from an embedded editable element (eg. an <input>
+                  // element inside a void node).
+                  if (relatedTarget === el) return
+
+                  if (isDOMElement(relatedTarget)) {
+                    // COMPAT: The event should be ignored if the focus is moving from
+                    // the editor to inside a void node's spacer element.
+                    if (relatedTarget.hasAttribute('data-slate-spacer')) return
+
+                    // COMPAT: The event should be ignored if the focus is moving to a
+                    // non- editable section of an element that isn't a void node (eg.
+                    // a list item of the check list example).
+                    const node = editor.toSlateNode(relatedTarget)
```

Comment:
This seems to cause a problem when clicking from an editor into another focusable element on the page. The `toSlateNode` function throws an error if the input is not in the editor. Should it just return null in that case (I believe mainline does this now for `findNode`)? If not, then` hasDomNode` seems to be needed to be checked first. This can be seen in the search highlight example.

Reply:
I was trying to see how far we can get with throwing, since it leads to less weirdness I think. Checking if it has it first sounds good to me.